src/backend/screencast.rs

```diff
@@ -0,0 +1,260 @@
+use std::sync::{Arc, Mutex};
+
+use async_trait::async_trait;
+use enumflags2::BitFlags;
+
+use crate::{
+    backend::{
+        request::{Request, RequestImpl},
+        session::{CreateSessionResponse, Session, SessionImpl, SessionManager},
+        MaybeAppID, MaybeWindowIdentifier, Result,
+    },
+    desktop::{
+        request::Response,
+        screencast::{CursorMode, SourceType, Streams as StartCastResponse},
+        HandleToken, PersistMode,
+    },
+    zvariant::{DeserializeDict, OwnedObjectPath, SerializeDict, Type},
+    AppID, PortalError, WindowIdentifierType,
+};
+
+#[derive(DeserializeDict, Type, Debug)]
+#[zvariant(signature = "dict")]
+pub struct CreateSessionOptions {}
+
+#[derive(DeserializeDict, Type, Debug)]
+#[zvariant(signature = "dict")]
+pub struct SelectSourcesOptions {
+    types: Option<BitFlags<SourceType>>,
+    multiple: Option<bool>,
+    cursor_mode: Option<CursorMode>,
+    restore_token: Option<String>,
```

Comment:
I was reading the spec and found that this field doesnt match the spec.

Shouldnt it be the following?

```suggestion
    restore_data: Option<(String, u32, Value<'static>)>,
```

Reply:
Indeed.

Reply:
Thanks for the quick reply.